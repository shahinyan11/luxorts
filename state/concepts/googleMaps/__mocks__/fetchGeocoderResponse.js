export default {
  data: {
    results: [
      {
        address_components: [
          {
            long_name: '240',
            short_name: '240',
            types: ['street_number'],
          },
          {
            long_name: 'North 7th Avenue',
            short_name: 'N 7th Ave',
            types: ['route'],
          },
          {
            long_name: 'Central City',
            short_name: 'Central City',
            types: ['neighborhood', 'political'],
          },
          {
            long_name: 'Phoenix',
            short_name: 'Phoenix',
            types: ['locality', 'political'],
          },
          {
            long_name: 'Maricopa County',
            short_name: 'Maricopa County',
            types: ['administrative_area_level_2', 'political'],
          },
          {
            long_name: 'Arizona',
            short_name: 'AZ',
            types: ['administrative_area_level_1', 'political'],
          },
          {
            long_name: 'United States',
            short_name: 'US',
            types: ['country', 'political'],
          },
          {
            long_name: '85007',
            short_name: '85007',
            types: ['postal_code'],
          },
        ],
        formatted_address: '240 N 7th Ave, Phoenix, AZ 85007, USA',
        geometry: {
          location: {
            lat: 33.4507026,
            lng: -112.0825682,
          },
          location_type: 'RANGE_INTERPOLATED',
          viewport: {
            northeast: {
              lat: 33.4520515802915,
              lng: -112.0812192197085,
            },
            southwest: {
              lat: 33.4493536197085,
              lng: -112.0839171802915,
            },
          },
        },
        place_id:
          'EiUyNDAgTiA3dGggQXZlLCBQaG9lbml4LCBBWiA4NTAwNywgVVNBIjESLwoUChIJ2dJpcCUSK4cR66Bm3xgAAh8Q8AEqFAoSCRvbefbPEiuHEaoczC_RXv9w',
        types: ['street_address'],
      },
    ],
    status: 'OK',
  },
};
