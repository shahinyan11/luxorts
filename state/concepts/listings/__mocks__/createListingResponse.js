export default {
  data: {
    data: {
      id: '83b0e92f-795b-4b3a-b527-0cd672dc9495',
      type: 'listing',
      attributes: {
        application_step: 'accomodation',
        status: 'in_draft',
      },
      relationships: {
        user_account: {
          data: {
            id: '06e67ade-80c5-46ef-8908-c14bad84f9fa',
            type: 'host_account',
          },
        },
        listing_property_type: {
          data: {
            id: '86b13618-08b3-42a3-8380-12b639411bca',
            type: 'listing_property_type',
          },
        },
        listing_location: {
          data: null,
        },
        listing_description: {
          data: null,
        },
      },
    },
    included: [
      {
        id: 'af39def7-be38-4027-9f06-a26d2d600044',
        type: 'property_type',
        attributes: {
          name: 'pariatur',
        },
      },
      {
        id: '86b13618-08b3-42a3-8380-12b639411bca',
        type: 'listing_property_type',
        attributes: {
          type_of_place: 'entire_room',
        },
        relationships: {
          property_type: {
            data: {
              id: 'af39def7-be38-4027-9f06-a26d2d600044',
              type: 'property_type',
            },
          },
        },
      },
      {
        id: '06e67ade-80c5-46ef-8908-c14bad84f9fa',
        type: 'host_account',
        attributes: {
          email: 'claud.bosco@runolfssonhalvorson.ca',
          public_id: 'RhGZjbl6',
          verified_status: 'phone_verified',
          status: 'active',
        },
        relationships: {
          user_profile: {
            data: null,
          },
        },
      },
    ],
  },
};
