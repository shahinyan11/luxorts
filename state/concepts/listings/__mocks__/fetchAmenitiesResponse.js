export default {
  data: {
    data: [
      {
        id: '5e066ad7-4194-4bd4-95ec-0d9715f8159b',
        type: 'amenity',
        attributes: {
          description: 'Maiores porro nostrum dolorum sunt.',
          name: 'Nolan Kertzmann',
          status: 'active',
          category: null,
        },
        relationships: {
          amenity_group: {
            data: {
              id: '6e9e46fd-d1d6-4a9a-920f-0360abc10bc6',
              type: 'amenity_group',
            },
          },
        },
      },
      {
        id: 'c277033f-1683-4d2d-a409-d00b0bbe8201',
        type: 'amenity',
        attributes: {
          description: 'Nemo magnam dolores quam voluptatibus distinctio unde.',
          name: 'Stephine Rohan',
          status: 'active',
          category: null,
        },
        relationships: {
          amenity_group: {
            data: {
              id: '1ae02fe9-0c6b-4e31-b1ae-7410e2bbc9ed',
              type: 'amenity_group',
            },
          },
        },
      },
    ],
    included: [
      {
        id: '6e9e46fd-d1d6-4a9a-920f-0360abc10bc6',
        type: 'amenity_group',
        attributes: {
          name: 'Amina Heaney',
        },
      },
      {
        id: '1ae02fe9-0c6b-4e31-b1ae-7410e2bbc9ed',
        type: 'amenity_group',
        attributes: {
          name: 'Torrie Sawayn',
        },
      },
    ],
  },
};
