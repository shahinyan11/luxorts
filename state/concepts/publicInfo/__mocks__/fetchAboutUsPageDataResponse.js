export default {
  data: {
    data: {
      data: null,
      meta: {
        properties_count: 0,
        cities_count: 0,
        hosts_count: 0,
      },
    },
  },
};
