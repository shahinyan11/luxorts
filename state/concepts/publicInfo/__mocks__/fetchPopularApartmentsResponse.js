export default {
  data: {
    data: [
      {
        id: '06a30f07-23b5-4900-b83c-b56986f119bc',
        type: 'listing',
        attributes: {
          bedrooms_amount: 1,
          baths_amount: 28,
          property_type_name: 'voluptatum',
          title: 'Alta Kling',
          guests_number: 3,
          beds_amount: 10,
          description: 'Vanessa Swift',
          type_of_place: 'shared_room',
          price_per_day: 6860000,
        },
        relationships: {
          limited_listing_photos: {
            data: [
              {
                id: 'bf4f23e4-bf5a-41f6-9bd3-7f18e3d55cf7',
                type: 'listing_photo',
              },
            ],
          },
        },
      },
    ],
    included: [
      {
        id: 'bf4f23e4-bf5a-41f6-9bd3-7f18e3d55cf7',
        type: 'listing_photo',
        attributes: {
          photo_urls: {
            small:
              'memory://listingphoto/bf4f23e4-bf5a-41f6-9bd3-7f18e3d55cf7/photo/small-21bf20aee591b473fd9bf94b1da2fdeb.jpg',
            medium:
              'memory://listingphoto/bf4f23e4-bf5a-41f6-9bd3-7f18e3d55cf7/photo/medium-d7964c233b11a9dc73ea331b80117253.jpg',
            original:
              'memory://listingphoto/bf4f23e4-bf5a-41f6-9bd3-7f18e3d55cf7/photo/56b59ca02c0db00b4d90f835501b8f97.jpg',
          },
          position: 1,
          description:
            'b5ln4wae0893f39lds2pnoc0cn8eorr3257at5h8yzvnq5o80zj0cfd5m99to629co23bc0vr9mrmdq9iyigdhobsq7idwy43aa95z9a7njgr62ya9v7lvjj3dkz3xqoazz5vha6ufi59mr3gdsdca5iorvqxmcjcfnrjxbknt8tbr4uvu64gj7a0miwtwufudprgtz700j4g5hkpgz1stjtn35dc5t8ghi7d18x7wx1r8xyw1qthqd8ip9fugr',
        },
      },
    ],
    links: {
      self: 1,
      last: 1,
    },
  },
};
