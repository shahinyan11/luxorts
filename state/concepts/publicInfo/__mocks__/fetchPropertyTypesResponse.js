export default {
  data: {
    data: [
      {
        id: 'd46e3bec-bb9a-4574-ba6d-1a3e19332b5a',
        type: 'property_type',
        attributes: {
          name: 'totam',
          listings_count: 1,
        },
      },
    ],
  },
};
