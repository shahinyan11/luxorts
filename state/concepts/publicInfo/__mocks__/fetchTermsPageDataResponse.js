export default {
  data: {
    data: {
      id: '0a16a67a-84bd-4140-887a-64033955d929',
      type: 'static_page',
      attributes: {
        updated_at: '2022-07-19T15:11:32.192Z',
        content: 'html',
        kind: 'terms_of_service',
      },
    },
  },
};
