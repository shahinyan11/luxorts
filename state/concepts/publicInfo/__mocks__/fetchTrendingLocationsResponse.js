export default {
  data: {
    data: [
      {
        id: 'united-states-pennsylvania-mineral-point',
        type: 'trending_location',
        attributes: {
          country: 'United States',
          city: 'Baltimore',
          state: 'Maryland',
        },
      },
    ],
  },
};
