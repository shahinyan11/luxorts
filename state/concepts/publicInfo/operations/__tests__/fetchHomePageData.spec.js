import { dispatch } from '__mocks__/react-redux';

import {
  fetchPopularApartments,
  fetchPropertyTypes,
  fetchTrendingLocations,
} from 'state/concepts/publicInfo/actions';
import fetchHomePageDataOperation from 'state/concepts/publicInfo/operations/fetchHomePageData';

describe('fetchHomePageData operation', () => {
  it('matches snapshot', () => {
    expect(fetchHomePageDataOperation).toMatchSnapshot();
  });

  it('dispatches actions', () => {
    fetchHomePageDataOperation.process(null, dispatch, jest.fn());

    expect(dispatch).toHaveBeenCalledWith(fetchTrendingLocations());
    expect(dispatch).toHaveBeenCalledWith(fetchPropertyTypes());
    expect(dispatch).toHaveBeenCalledWith(fetchPopularApartments());
  });
});
