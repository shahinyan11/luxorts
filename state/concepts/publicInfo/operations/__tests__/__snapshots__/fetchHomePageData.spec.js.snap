// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchHomePageData operation matches snapshot 1`] = `
Object {
  "cancelType": undefined,
  "debounce": 0,
  "latest": true,
  "name": undefined,
  "process": [Function],
  "processOptions": Object {
    "dispatchMultiple": true,
  },
  "throttle": 0,
  "transform": undefined,
  "type": "publicInfo/FETCH_HOME_PAGE_DATA",
  "validate": undefined,
  "warnTimeout": 60000,
}
`;
